### PR TITLE
add support for loongarch64 

### PR DESCRIPTION
**Pull request by yzewei**
I'm very sorry, due to my operational errors, the last PR failed, re -submitted to complete the Loongarch support work.
The LoongArch architecture (LoongArch) is an Instruction Set Architecture (ISA) that has a RISC style.
@crazy-max @tonistiigi

### CODE REVIEW

**Comment by tonistiigi** [comment]
Add testcases for this in https://github.com/tonistiigi/binfmt/blob/master/.github/workflows/ci.yml#L114-L132 as well.

@crazy-max Is there anything else that needs to be enabled for CI side?

**Comment by yzewei** [comment]
> Add testcases for this in https://github.com/tonistiigi/binfmt/blob/master/.github/workflows/ci.yml#L114-L132 as well.
> 
> @crazy-max Is there anything else that needs to be enabled for CI side?

Add debian image running test with the Loongarch architecture

**Comment by yzewei** [comment]
@crazy-max Thanks!

**Comment by crazy-max** [comment]
> @crazy-max Is there anything else that needs to be enabled for CI side?

I don't think so :+1:

**Comment by crazy-max** [comment]
Seems test fails for loong64: https://github.com/tonistiigi/binfmt/actions/runs/7271830266/job/19818345215?pr=151#step:6:138

I don't see `linux/loong64` in the list of supported platforms either: https://github.com/tonistiigi/binfmt/actions/runs/7271830266/job/19818345215?pr=151#step:6:50

**Comment by yzewei** [comment]
> Seems test fails for loong64: https://github.com/tonistiigi/binfmt/actions/runs/7271830266/job/19818345215?pr=151#step:6:138
> 
> ```
> uname: error while loading shared libraries: libc.so.6: cannot stat shared object: Error 38
> ```
> 
> ~I don't see `linux/loong64` in the list of supported platforms either: https://github.com/tonistiigi/binfmt/actions/runs/7271830266/job/19818345215?pr=151#step:6:50~

Change the running test

**Review by crazy-max — APPROVED** [review]
LGTM

**Comment by crazy-max** [comment]
@yzewei Can you squash your commits please?

**Comment by yzewei** [comment]
> @yzewei Can you squash your commits please?

OK

File: scripts/configure_qemu.sh [hunk 1]
```diff
@@ -34,6 +34,9 @@ if [ -z "$QEMU_TARGETS" ]; then
   if [ "$arch" != "mips64" ] ; then
     QEMU_TARGETS="$QEMU_TARGETS mips64-linux-user"
   fi
+  if [ "$arch" != "loongarch64" ] ; then
```
Comment (tonistiigi) [inline scripts/configure_qemu.sh:37]: This is the `GOARCH` formatting, so should be loong64.

Comment (yzewei) [reply scripts/configure_qemu.sh:37]: OK

**Comment by crazy-max** [comment]
@yzewei I opened #153 to check qemu build and also adds missing tests in https://github.com/tonistiigi/binfmt/commit/82f3ca76eeaf414399633786ca2c07e88c169514

**Comment by yzewei** [comment]
> @yzewei I opened #153 to check qemu build and also adds missing tests in [82f3ca7](https://github.com/tonistiigi/binfmt/commit/82f3ca76eeaf414399633786ca2c07e88c169514)

It looks like to adapt xx first

**Review by tonistiigi — APPROVED** [review]

**Pull request merged**